### PR TITLE
[WIP] Support sdk remote workflow

### PR DESCRIPTION
**Pull request by ckiosidis**
# TL;DR

Adding support for SdkRemoteWorkflow, similarly to the existing SdkRemoteTask

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Created a SdkRemoteWorkflow class that implements the apply method to return a SdkNode.
The SdkWorkflowNode/WorkflowNode classes were already implemented previously.

## Tracking Issue
Fixes https://github.com/flyteorg/flyte/issues/1002

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteWorkflow.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        .domain(domain)
+        .project(project)
+        .name(name)
+        .inputs(inputs)
+        .outputs(outputs)
+        .build();
+  }
+
+  @Override
+  public SdkNode apply(
```
[the remark this answers is not in the document]
Comment (ckiosidis) [reply flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteWorkflow.java:65]: it's not, probably need to hydrate the SdkWorkflow

**Comment by ckiosidis** [comment]
Let's discuss this PR early next year, maybe we don't need to merge it and stick to the explicit SdkRemoteLaunchPlan

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteWorkflow.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ckiosidis) [reply flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteWorkflow.java:64]: It's not. SdkRemoteWorkflows are not really used to create WorkflowNodes.
The flytekit(python) sdk fetches the default launch plan when it's a workflow and it uses that one.

**Comment by ckiosidis** [comment]
The SdkRemoteWorkflow is just a wrapper of the default launch plan for that workflow https://github.com/flyteorg/flytekit/blob/ab0489112624d85a1341073eb722fa328e008745/flytekit/remote/remote.py#L845
Since the SdkRemoteLaunchPlan is implemented separately, maybe we don't need to implement this one

**Comment by ckiosidis** [comment]
We decided to not implement the SdkRemoteWorkflow for now.
In Flyte, the Tasks and LaunchPlans are the entities that are supposed to be used remotely.
(For flytekit python) Although the FlyteWorkflow is implemented in the FlyteRemote, it's just [fetching the default launch plan](https://github.com/flyteorg/flytekit/blob/ab0489112624d85a1341073eb722fa328e008745/flytekit/remote/remote.py#L866) for the workflow and executing it.

Use the [SdkRemoteLaunchPlan](https://github.com/flyteorg/flytekit-java/blob/master/flytekit-java/src/main/java/org/flyte/flytekit/SdkRemoteLaunchPlan.java) for Remote LaunchPlan or Workflow executions.
